src/utils/transactionReceipt.ts

```diff
@@ -0,0 +1,114 @@
+/* eslint-disable no-nested-ternary */
+import {
+  GetTransactionReceiptResponseWoHelper,
+  RejectedTransactionReceiptResponse,
+  RevertedTransactionReceiptResponse,
+  SuccessfulTransactionReceiptResponse,
+  TransactionExecutionStatus,
+} from '../types';
+import type {
+  TransactionReceiptCallbacks,
+  TransactionReceiptCallbacksDefault,
+  TransactionReceiptStatus,
+  TransactionReceiptUtilityInterface,
+  TransactionReceiptValue,
+} from '../types/transactionReceipt';
+
+/**
+ * Utility that analyses transaction receipt response and provides helpers to process it
+ * @example
+ * ```typescript
+ * const responseTx = new ReceiptTx(receipt);
+ * responseTx.match({
+ *   success: (txR: SuccessfulTransactionReceiptResponse) => { },
+ *   rejected: (txR: RejectedTransactionReceiptResponse) => { },
+ *   reverted: (txR: RevertedTransactionReceiptResponse) => { },
+ *   error: (err: Error) => { },
+ * });
+ * responseTx.match({
+ *   success: (txR: SuccessfulTransactionReceiptResponse) => { },
+ *   _: () => { },
+ * }
+ * ```
+ */
+export class ReceiptTx implements TransactionReceiptUtilityInterface {
+  public readonly statusReceipt: TransactionReceiptStatus;
+
+  public readonly value: TransactionReceiptValue;
+
+  constructor(receipt: GetTransactionReceiptResponseWoHelper) {
+    [this.statusReceipt, this.value] = ReceiptTx.isSuccess(receipt)
+      ? ['success', receipt]
```

Comment:
'success' and rest of the statuses are repeated as magic strings also below on line 71 etc... maybe to extract them in a constant?

Reply:
The type TransactionReceiptStatus is listing these strings.
![image](https://github.com/starknet-io/starknet.js/assets/81040730/f9113e73-57c4-41d3-b16c-cc499af6d49b)
@penovicp Is it possible to create an enum based on the strings listed in a type?